### ISSUE TITLE
Bug: NullPointerException using Foreign key display values

### ISSUE THREAD

**Issue by ChrisCalculus** [metabase/metabase#7346]
Hi,

I've got an issue with setting up a new metabase env. Running into a NullPointerException on one of the tables.

- Your browser and the version: Safari 11.03
- Your operating system: OS X 10.13.3
- Your databases: Postgres (Google Cloud SQL)
- Metabase version: 0.28.6
- Metabase hosting environment: Kubernetes
- Metabase internal database: Postgres (Google Cloud SQL)

### Steps to reproduce:
- Add source db with 2 tables (i.e. tables: Issues & Users)
- Set 2 foreign key mappings in the Data model:
  - Issue.Author_ID to PK User.ID
  - Issue.Assignee_ID to PK User.ID
- Set display value to 'Foreign Key > User.Name' on both

Weird enough, when changing the display value to original on the Issue.Assignee_ID it works. Doing this for Author_ID has no influence.

The result is the NullPointerException (see stacktrace below):  

```Java
Apr 13 16:09:36 WARN metabase.query-processor :: {:status :failed, :class java.lang.NullPointerException, :error "java.lang.NullPointerException", :stacktrace ["query_processor.middleware.add_dimension_projections$col__GT_dim_map.invokeStatic(add_dimension_projections.clj:71)" "query_processor.middleware.add_dimension_projections$col__GT_dim_map.invoke(add_dimension_projections.clj:64)" "query_processor.middleware.add_dimension_projections$remap_results.invokeStatic(add_dimension_projections.clj:127)" "query_processor.middleware.add_dimension_projections$remap_results.invoke(add_dimension_projections.clj:114)" "query_processor.middleware.expand$expand_middleware$fn__29769.invoke(expand.clj:601)" "query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__28153.invoke(add_row_count_and_status.clj:14)" "query_processor.middleware.driver_specific$process_query_in_context$fn__29900.invoke(driver_specific.clj:12)" "query_processor.middleware.resolve_driver$resolve_driver$fn__31353.invoke(resolve_driver.clj:15)" "query_processor.middleware.cache$maybe_return_cached_results$fn__28621.invoke(cache.clj:146)" "query_processor.middleware.catch_exceptions$catch_exceptions$fn__29817.invoke(catch_exceptions.clj:58)" "query_processor$process_query.invokeStatic(query_processor.clj:130)" "query_processor$process_query.invoke(query_processor.clj:126)" "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:241)" "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:235)" "query_processor$fn__31387$process_query_and_save_execution_BANG___31392$fn__31393.invoke(query_processor.clj:281)" "query_processor$fn__31387$process_query_and_save_execution_BANG___31392.invoke(query_processor.clj:267)" "query_processor$fn__31411$process_query_and_save_with_max_BANG___31416$fn__31417.invoke(query_processor.clj:302)" "query_processor$fn__31411$process_query_and_save_with_max_BANG___31416.invoke(query_processor.clj:298)" "api.dataset$fn__37935$fn__37938$fn__37939.invoke(dataset.clj:49)" "api.common$fn__20323$invoke_thunk_with_keepalive__20328$fn__20329$fn__20330.invoke(common.clj:402)"], :query {:type "query", :query {:source_table 5}, :parameters [], :constraints {:max-results 10000, :max-results-bare-rows 2000}, :info {:executed-by 1, :context :ad-hoc, :card-id nil, :nested? false, :query-hash [108, 84, 50, 34, 11, 38, 119, 50, -56, -60, 42, 94, 101, -64, 90, 33, 112, 60, 30, 100, 92, 60, -86, -3, -114, -10, -114, -29, 62, -24, -76, -68], :query-type "MBQL"}}, :expanded-query nil}  

Apr 13 16:09:36 WARN metabase.query-processor :: Query failure: java.lang.NullPointerException ["query_processor$assert_query_status_successful.invokeStatic(query_processor.clj:209)" "query_processor$assert_query_status_successful.invoke(query_processor.clj:202)" "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:242)" "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:235)" "query_processor$fn__31387$process_query_and_save_execution_BANG___31392$fn__31393.invoke(query_processor.clj:281)" "query_processor$fn__31387$process_query_and_save_execution_BANG___31392.invoke(query_processor.clj:267)" "query_processor$fn__31411$process_query_and_save_with_max_BANG___31416$fn__31417.invoke(query_processor.clj:302)" "query_processor$fn__31411$process_query_and_save_with_max_BANG___31416.invoke(query_processor.clj:298)" "api.dataset$fn__37935$fn__37938$fn__37939.invoke(dataset.clj:49)" "api.common$fn__20323$invoke_thunk_with_keepalive__20328$fn__20329$fn__20330.invoke(common.clj:402)"]

```

**Comment by senior** [comment]
@ChrisCalculus I'm not able to reproduce your issue. I created two tables like you've described and it works for me.
```
create table "user" (id integer primary key, name text);
create table issue (id integer primary key, author_id integer references "user" (id), assignee_id integer references "user" (id), description text);
```
I then set `assignee_id` and `author_id` to foreign key, and their remapping to use the name column from user, seems to work ok.

Can you try to recreate your issue with a minimal setup like I have and then paste in your schema/steps?

**Comment by MarkRx** [comment]
I have also seen this issue. We are not sure what is needed to get Metabase into this state. When I had it reproduced in sandbox after tinkering it disappeared after I did a full database sync. Otherwise it seems to randomly show up.

**Comment by binaryduke** [comment]
Also seeing this issue and it is a killer. We have 3 tables:
Sale
Product
Product Category

Sale holds product_id as a foreign key.
Product holds product_category_id as a foreign_key

Running a query on sale that references the product_category_id foreign key via the product table returns the same error.

**Comment by Crusader4Christ** [comment]
Same error

**Comment by jike212** [comment]
Same issue here as well with mapping foreign keys other values

**Comment by mazameli** [comment]
@jike212 reported that 0.30.4 fixed this issue, but @binaryduke said it was still present in 0.30.4. Are you folks still seeing this issue in version 0.31.x?

**Comment by mbvictoria** [comment]
Closing - has not been reported in recent versions, was reported to be fixed in 0.30.4
